File: src/client/src/models/PushSubscription.ts
import { PushSubscriptionKeys } from "./PushSubscriptionKeys";

export class PushSubscription {
    userId: number;
    endpoint: string;
    keys: PushSubscriptionKeys;

    constructor(userId: number, endpoint: string, keys: PushSubscriptionKeys) {
        this.userId = userId;
        this.endpoint = endpoint;
        this.keys = keys;
    }
}

File: src/jamera.api/Jamera/Models/PushSubscription.cs
using System.ComponentModel.DataAnnotations;

namespace Jamera.Models;

public class PushSubscription
{
    
    [Key]
    public Guid Id { get; set; }
    public Guid UserID { get; set; }
    public string Endpoint { get; set; }
    public PushSubscriptionKeys Keys { get; set; }
}